mdal/frmts/mdal_mike21.cpp

```diff
@@ -128,6 +128,39 @@ bool MDAL::DriverMike21::canReadMesh( const std::string &uri )
   {
     return false;
   }
+
+  in.clear();
+  in.seekg( 0, std::ios::beg );
+
+  size_t vertexCount = getVertexCount( line );
+  size_t faceCount = 0;
+
+  size_t lineNumber = 0;
```

Comment:
I do not like this approach , if the file is massive you pretty much read/parse it fully only to say it is MIKE 21 file... 

canReadMesh is called many times for each file (even read by other drivers) so it should be fast